src/components/CodeEditor/CodeEditor.js

```diff
@@ -282,7 +312,7 @@ class CodeEditor extends Component {
   }
 
   _onChange = code => {
-    this.setState(this._updateState(code));
+    this.setState(this._updateState(code, this.state.showJSX));
```

Comment:
We should change this to:

```js
this.setState(state => this._updateState(code, state.showJSX));
```

The benefit of using a callback for `setState` is that it avoids references potentially stale values in `this.state`. This didn't matter before because we weren't references `this.state` but now that we are....we should use the callback form. 😄